codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonShapeDeserVisitor.java

```diff
@@ -300,4 +281,36 @@ private void writeJsonTypeAssertStub(GoWriter writer, Shape shape) {
         writer.openBlock("if !ok {", "}", () -> writer.write("return fmt.Errorf(\"unexpected JSON type %v\", value)"));
         writer.write("");
     }
+
+    public static final class Builder implements SmithyBuilder<JsonShapeDeserVisitor> {
```

Comment:
Our typical pattern has been to make the Builder class private to the top-level class, and the top level class provide a static builder method that returns a new builder. That's consistent with how I believe most/all the Smithy builders are in their implementations.

Reply:
will take a look at those and duplicate that pattern